### ISSUE TITLE
Can't install Java and Python package using the module

### ISSUE THREAD

**Issue by jordigg** [TheKevJames/puppet-homebrew#32]
I can't install the java package using brew cask. I get the following error:

```
Error: /Stage[main]/Profiles::Common::Darwin/Package[java]: Could not evaluate: invalid number of elements (4 for 1..2)
```

If I run manually `brew cask install java` works just fine.

Another error I've noticed is that Python doesn't get installed, maybe because detects that mac already have another Python installed but again, doing a `brew install python` works just fine. Is just when using puppet with this modules that doesn't work.

Does anyone else have the same problem? I'm trying on fresh installed MacMinis running el Capitan with all the latest updates/Xcode.

**Comment by zbentley** [comment]
I am also being affected by this issue. It has occurred in two situations:

- On a clean install of El Capitan without `csrutil disable` having been run.
- On other installs of El Capitan (including the clean install) on `cask`-provided packages with `install_options` set to contain `--appdir=/Applications` or similar.

The first correllation (`csrutil disable`) may be false; there were other things wrong with that system.

The workaround for me has been to stop using install_options, though that is far from ideal.

**Comment by TheKevJames** [comment]
Hey, I added a test for this on #33 but it seems to be passing. I'm currently looking into if this is an issue specific to El Cap... Unfortunately, I don't actually have access to a Mac for the next couple months, so my best bet is to dig deeper into TravisCI matrix testing. If anyone can find a test case that fails on CI or can track down what's going wrong, that would be much appreciated.

**Comment by zbentley** [comment]
I'll try to give repro a shot on travis next weekend, but may not have time. Thanks for the quick responses!

**Comment by zbentley** [comment]
I found out how to reproduce this. It only occurs if `brew cask` is partially or completely missing from the system, _or has never been used_, before the first puppet run.

**To work around:**
Manually install one (any one) program via `brew cask` before invoking Puppet; puppet-managed cask programs should work after that.


**To reproduce:**
Steps below work on multiple 10.11 systems. Steps 1 through 3 can be skipped if a system is clean provisioned/brand new (though I'm not sure if Travis systems use cask internally or not).

1. Remove all homebrew-cask apps either via `brew cask zap` or `brew cask uninstall --force`. You should be able to do this in-place on your workstation without damaging any of your prefs or user data with `for app in $(brew cask list); do brew cask uninstall --force $app; done` or similar.
2. Remove homebrew-cask by untapping it and all caskroom subtaps. 
3. Remove the caskroom directory entirely from your computer.
4. Reinstall homebrew-cask/do `brew cask cleanup` to verify that it's alive.
5. Try to install any package with an (implicit/fallback or explocit) `brewcask` provider via Puppet. The `invalid number of elements (4 for 1..2)` error should occur.

It's probably an edge case with missing directories or missing cask configs or something. Maybe a list gets flattened out to a missing argument instead of left empty if there's nothing in the caskroom/the caskroom doesn't exist?

**Comment by jordigg** [comment]
Today I was just playing again with it on new machines and still got the same error. It happens with any application I want to install from brew cask. I have to install any manually and then the module will install the rest. Without that manual step it will always fail.

**Comment by jordigg** [comment]
Got a solution and sent a PR. Seems that python should have installed just fine, the problem was just brewcask. Have sent a fix on #39 please check and merge if works for you too. Have tested on a new machine running El Capitan.

**Comment by TheKevJames** [comment]
Hey! Should be fixed with @jordigg 's PR (released in [1.4.2](https://forge.puppet.com/thekevjames/homebrew/readme)). If the problem persists, please let me know and reopen the issue.

On a personal level, sorry for taking so long to respond to this @zbentley

**Comment by zbentley** [comment]
Thanks for the merge and update! I can confirm that @jordigg's PR worked; I merged it into my local fork and tested recently, though my tests were on a non-pristine environment (I tried to remove all traces of homebrew first). With moderate certainty I agree that this is closed.

And on a personal level: you're doing this for free, no worries!

**Comment by katzenbaer** [comment]
I am on 1.4.2, and installing Python still gives me:
```
Error: /Stage[main]/Main/Node[hostname.local]/Package[python]: Could not evaluate: invalid number of elements (4 for 1..2)
```

**Comment by jordigg** [comment]
Works for me on a few brand new systems, is yours a nee one? Which user do you run brew under?
I'll try to send a new PR today, there's a better way for fixing these messages.
Could you copy the outup using `puppet agent -t --debug --verbose` related to this error?

**Comment by jordigg** [comment]
@katzenbaer sent another fix that should fix all this errors #50
